<?php

namespace tests\eLife;

use PHPUnit\Framework\TestCase;

class messageConversionTest extends TestCase
{
    /**
     * @test
     * @dataProvider providerAnchorTagsToPlainUrls
     */
    public function it_will_add_anchor_tags_to_plain_urls($raw_message, $expected = null)
    {
        $actual = add_anchor_tags_to_plain_urls($raw_message);
        $this->assertEquals($expected ?? $raw_message, $actual);
    }

    public function providerAnchorTagsToPlainUrls()
    {
        yield 'no urls' => [
            "Not a url in sight!!",
        ];
        yield 'only anchored urls' => [
            "<a href=\"http://bbc.co.uk\">BBC</a>\n\nSome text <a href=\"http://google.com\">http://google.com</a>.",
        ];
        yield 'only plain urls' => [
            "http://bbc.co.uk\n\nSome text http://google.com.",
            "<a href=\"http://bbc.co.uk\">http://bbc.co.uk</a>\n\nSome text <a href=\"http://google.com\">http://google.com</a>.",
        ];
        yield 'mix of anchored and plain urls' => [
            "<a href=\"http://bbc.co.uk\">BBC</a>\n\nSome text http://google.com;",
            "<a href=\"http://bbc.co.uk\">BBC</a>\n\nSome text <a href=\"http://google.com\">http://google.com</a>;",
        ];
        yield 'plain url followed by semi-colon' => [
            "[Originally posted on 18 September 2014]\n\na) There are two independent 3D reconstructions of the MAVS filaments: one was published in eLife (2014 Jan. 1; 3:e01489) by Xu et al, and the other in Molecular Cell (2014 Aug. 21; 55(4):511-23) by Wu et al.\n\nb) We have uploaded our original micrograph images and detector images and raw pre-boxed filament segments to EMPIAR (http://pdbe.org/empiar; accession code: EMPIAR-10014). Those interested may analyze the data themselves. Please refer to our detailed descriptions on analyzing the individual datasets separately before scaling them based on the axial rises and merging them together.\n\nc) In the following paragraphs, we would like to submit an analysis of the differences between these two studies. The comparison of our experiments and data with those by Wu et al reveals significant differences in the specimens, the data and the results from 3D reconstructions. We will explain the differences from the following six aspects.\n\n1) The MAVS filaments in the two papers were prepared very differently. In the Wu et al paper, the 10 kDa MAVS CARD P2A mutant protein was expressed as a fusion protein (~35 kDa) in bacteria and purified and formed filaments through a denaturation-renaturation process. The resultant filament was treated with proteases to remove His-tag and SNAP-domain (~20 kDa), leaving 19 amino acid residues (PGYQDSVDEALREAQTNSA) to the N-terminus of the MAVS CARD P2A mutant. There was no evidence in this or other papers that the renatured filaments prepared in such ways had the ability to activate endogenous MAVS or promote IRF3 dimerization, which are the functional readouts of the signaling pathway.\n\nIn contrast, the specimens in Xu et al paper in eLife were prepared from purified Flag-tagged wild-type MAVS CARD expressed in HEK293T cells. The protein was purified to apparent homogeneity and formed filaments spontaneously without going through a denaturation-renaturation process. Importantly, the filaments that we prepared this way potently activated endogenous MAVS and promoted IRF3 dimerization in a cell-free assay we established (Hou et al., Cell, 2011). Thus, although our MAVS CARD filaments prepared from the native protein appeared less rigid than the engineered filaments shown in the Wu et al paper, our filaments were biochemically functional.\n\n2) The difference in the raw images can be examined by comparing the summed power spectra from the boxed particle images (Fig. 1). There are two clear differences. First, the power spectrum from Wu et al showed that Layerline 9 (ll9) is split around the vertical axis (helical axis), suggesting a C1 assembly (Fig. 1A). The power spectrum of our filaments (Fig. 1B) is clearly different. We found that the ll9 is meridian, suggesting an axial rise of ~16.7 Ã… for our filament. Please note that layerline splitting could also be caused by filament tilting, especially when filaments were crowded together. Secondly, the Layerline 1 (ll1) in Fig 1A (from Wu et al data) is almost invisible. In contrast, the ll1 peaks in our spectrum (Fig. 1B, Xu et al data) are clearly visible. Considering that the relative intensity of Layerlines 4 & 5 in Fig. 1A is stronger than the corresponding ones in Fig 1B, the ll1 peaks in Fig 1B would have completely disappeared if our filaments took the same C1 symmetry as those by Wu et al. The pattern of the layerlines in Fig 1B is similar to that of the C3 helical filament formed by the ASC Pyrin Domain (Fig. 1C), which was recently published by Lu et al (Cell 156, 1103). In particular, the ASC Pyrin filaments also showed clear layerline 1 peaks, similar to our MAVS CARD filaments. Like the MAVS CARD domain, Pyrin domain is a member of the Death Domain superfamily. Recent studies have demonstrated that both MAVS CARD and ASC Pyrin domains behave as prion-like filaments (Cai et al. 2014, Cell 156,1207; Lu et al. 2014, Cell 156, 1103). Our analyses suggest that MAVS CARD and ASC Pyrin domains both adopt C3 symmetry when they form prion-like helical filaments.\n\nTo independently evaluate our raw data, we invited two experts from two different institutions to calculate the power spectra from a good set of boxed segments using their own software packages. Their results (Fig. 1D and 1E) were essentially the same as ours as shown in Fig. 1B, and the peaks in Layerline 1 can be recognized despite that the signal-to-noise ratio in our raw data was lower than that of Wu et al. Layerline 1 reflects the long-range property of the filaments, and this difference in the power spectra is critical in building the filament model.\n\nTherefore, three different groups independently arrived at the same power spectrum based on the raw dataset of our MAVS CARD filament, which was presented in our eLife paper. This power spectrum is very similar to that of ASC Pyrin filaments, which exhibit the C3 symmetry, and is clearly different from the power spectrum of the renatured MAVS CARD filament presented in the Wu et al paper, which shows C1 symmetry. In particular, the two clear differences in power spectra (layerlines 1 and 9) indicate that the two filaments have different symmetry properties.\n\n3) Because of the potential ambiguity in the Fourier-Bessel indexing, especially with the limited resolution of our data, we had tested systematically the C1, C2, C3 and C4 symmetry against our data. Our test of C1 around the parameter sets of (102Âº, 5.0-5.6 Ã…) did not show any sign of convergence. Now with the symmetry property of Wu et al, we compared our test results with their C1 symmetry (101.1Âº, 5.13 Ã…; black â€œ+â€ in Fig. 2A). Our tests started from different positions (highlighted by black circles), but none of them converged to the symmetry parameters of Wu et alâ€™s filament. More strikingly, even when we started from the C1 parameters (brown squares), the refinement went away immediately, suggesting that our data are inconsistent with the C1 symmetry of Wu et al.\n\nIn contrast, when we tested the convergence around the C3 parameters (53.55Âº, 16.75 Ã…; black + in Fig 2B), all tests from different starting points (highlighted by black circles in Fig. 2B) converged to similar parameters. The convergence tolerated at least 10% perturbation in axial rise and at least ~3% perturbation in rotational angles. These tests argue strongly that our data are better described by the C3 symmetry, not the C1 symmetry as in the filaments by Wu et al.\n\nAs a control test, the C3 symmetry is a combination of a C1 and a three-fold rotation around the helical axis. It means that a C1 search around (53.55Âº, 16.7 Ã…) against our data should lead to a stable solution, and the obtained 3D reconstruction should have apparent three-fold rotational symmetry. These two predictions were indeed true. The C1 search of our data starting with a cylindrical volume stabilized at (53.02Âº, 16.60 Ã…), which are within 2% error range around the C3 parameters, and the obtained map showed an apparent three fold symmetry. This control test verified two important aspects of our analysis: 1) our C1 search procedure is reliable; 2) the three-fold symmetry appears intrinsic to our data at the current resolution.\n\n4) Some of the raw cryoEM images of our filaments taken at a high defocus level showed lower density in the middle (for example, red arrows in Fig. 3A in the attached PDF). This feature suggests that there might be a pore in the middle of the filament. As a test, we randomly assigned angles to the images aligned against a cylinder, and obtained an averaged reconstruction. The density showed a pore (~16 Ã… in diameter) in the middle (Fig. 3B). This feature was again observed in the final reconstruction of our filament, and was seen in the electron tomogram of the filaments.\n\nIn contrast, the C1 filament, whose two views are shown (Fig. 3C) as in the Wu et al Molecular Cell paper, is densely packed, and does not harbor a sizeable pore in the middle.\n\nThus, the presence of the central pore in our filament is compatible with the C3 reconstruction model presented in our eLife paper, but not the C1 model in the Wu et al paper.\n\n5) In the Mol Cell paper by Wu et al, the proximity of W56 and R43 was used as a piece of key evidence to discriminate the two structural models (Fig. 2B in the Wu et al paper). Even though the electrostatic compensation between W56R and R43D/E worked for in vitro filament formation by the seeding/ denaturation/ renaturation process (Fig. 2D in the Wu et al paper), the IFNbeta reporter assay showed that these double mutations did not rescue the functions of either human or duck MAVS in cells (Fig. S2D and S2F in the Wu et al paper). Thus, this experiment does not support that the C1 filament is a functional form capable of triggering MAVS signaling in vivo.\n\n6) It is well known that it is difficult to detect a stable complex between RIG-I and MAVS, and there is strong evidence to suggest that RIG-I dissociates from MAVS after MAVS starts to form the filaments; this allows RIG-I to promote MAVS polymerization in a â€˜catalyticâ€™ manner. In our model, the symmetry mismatch between the RIG-I tandem CARD tetramer and the C3 helix suggests that after the initiation of the filament, the RIG-I active complex may fall off, instead of staying bound. The molecular imprinting mechanism suggested by Wu et al was based in part on the complex of the RIG-I tandem CARD and the MAVS CARD mutant that were covalently linked together. The C1 model suggests that RIG-I CARDs might stay bound to MAVS after the filament formation.\n\nIn summary, our extensive analyses reconfirm that the filaments formed by the native wild-type MAVS CARD domain exhibit C3 symmetry, which is structurally different from the C1 filaments prepared through a denaturation-renaturation process as presented in the Wu et al paper. Future research should determine whether one or both of these structures represent the physiologically activated form of MAVS in vivo.\n\nFigure legends:\n\nFigure 1. Comparison of the summed power spectra from the raw data A) Power spectrum of the C1 filaments in Wu et al. (Mol Cell) B) Power spectrum of the C3 filaments in Xu et al. (eLife) C) Power spectrum of one of the sorted bin of images from the Cell paper by Lu et al 2014 on the ASC PYD filament, which shows C3 symmetry. Sorting sharpened the layerline 1 here. The pattern is similar to B, not A, because the layerline 1 peaks are almost invisible in A. D & E) Power spectra calculated by two independent scientists respectively; these spectra are consistent with B, supporting the model presented in the eLife paper by Xu et al.\n\nFigure 2. Convergence test of C1 versus C3 against the image data from filaments made of native wild-type MAVS CARD proteins A) C1 test. The starting points, highlighted by black circles, are all within the vicinity of the symmetry of the Molecular Cell paper by Wu et al (101.1Âº, 15.1Ã…). The refinements all led to further deviation from the expected C1 symmetry parameters. B) C3 test. The starting points are highlighted by black circles surrounding the first markers. The refinements all converged to the vicinity of (53.55Âº and 16.75 Ã…).\n\nFigure 3. A central pore in the filament made from native MAVS CARD proteins A) The cryoEM image from a short segment of the CARD filament showing the middle (red arrows) of the filament has lower density than the two edges. B) A reconstruction calculated from filaments that were aligned against a cylinder reference volume and assigned with random rotational angles along helical axes. Because of the averaging the volume is solid, except that it has lower density in the middle portion. The current threshold level gives a mass that is about 1.5 folds of the final reconstruction. At this level, there is a 16 Ã… hole in the middle, consistent with a lower density level in the middle of the raw images in A. C) Two views of the C1 filament from the Molecular Cell paper by Wu et al. The right panel is in the same orientation as in B. The densely packed units should display a higher density in the central part of the filament, which is distinct from the filament shown in A.\n\n\nhttps://uploads.disquscdn.com/images/1e5b6a8f7d686ad8583f41c5ea7c0a8731bb0b34f489370b6026ae6819bda891.jpg  https://uploads.disquscdn.com/images/0a865cd1e8c92a92047cc97f3ba8852bfe486d2f5d64b9dde0398c6b83e87109.jpg  https://uploads.disquscdn.com/images/3c954ee3404c1de29fbeaacc5c47d539cb5756397ad16f5b80416598d26c57c5.jpg",
            "[Originally posted on 18 September 2014]\n\na) There are two independent 3D reconstructions of the MAVS filaments: one was published in eLife (2014 Jan. 1; 3:e01489) by Xu et al, and the other in Molecular Cell (2014 Aug. 21; 55(4):511-23) by Wu et al.\n\nb) We have uploaded our original micrograph images and detector images and raw pre-boxed filament segments to EMPIAR (<a href=\"http://pdbe.org/empiar\">http://pdbe.org/empiar</a>; accession code: EMPIAR-10014). Those interested may analyze the data themselves. Please refer to our detailed descriptions on analyzing the individual datasets separately before scaling them based on the axial rises and merging them together.\n\nc) In the following paragraphs, we would like to submit an analysis of the differences between these two studies. The comparison of our experiments and data with those by Wu et al reveals significant differences in the specimens, the data and the results from 3D reconstructions. We will explain the differences from the following six aspects.\n\n1) The MAVS filaments in the two papers were prepared very differently. In the Wu et al paper, the 10 kDa MAVS CARD P2A mutant protein was expressed as a fusion protein (~35 kDa) in bacteria and purified and formed filaments through a denaturation-renaturation process. The resultant filament was treated with proteases to remove His-tag and SNAP-domain (~20 kDa), leaving 19 amino acid residues (PGYQDSVDEALREAQTNSA) to the N-terminus of the MAVS CARD P2A mutant. There was no evidence in this or other papers that the renatured filaments prepared in such ways had the ability to activate endogenous MAVS or promote IRF3 dimerization, which are the functional readouts of the signaling pathway.\n\nIn contrast, the specimens in Xu et al paper in eLife were prepared from purified Flag-tagged wild-type MAVS CARD expressed in HEK293T cells. The protein was purified to apparent homogeneity and formed filaments spontaneously without going through a denaturation-renaturation process. Importantly, the filaments that we prepared this way potently activated endogenous MAVS and promoted IRF3 dimerization in a cell-free assay we established (Hou et al., Cell, 2011). Thus, although our MAVS CARD filaments prepared from the native protein appeared less rigid than the engineered filaments shown in the Wu et al paper, our filaments were biochemically functional.\n\n2) The difference in the raw images can be examined by comparing the summed power spectra from the boxed particle images (Fig. 1). There are two clear differences. First, the power spectrum from Wu et al showed that Layerline 9 (ll9) is split around the vertical axis (helical axis), suggesting a C1 assembly (Fig. 1A). The power spectrum of our filaments (Fig. 1B) is clearly different. We found that the ll9 is meridian, suggesting an axial rise of ~16.7 Ã… for our filament. Please note that layerline splitting could also be caused by filament tilting, especially when filaments were crowded together. Secondly, the Layerline 1 (ll1) in Fig 1A (from Wu et al data) is almost invisible. In contrast, the ll1 peaks in our spectrum (Fig. 1B, Xu et al data) are clearly visible. Considering that the relative intensity of Layerlines 4 & 5 in Fig. 1A is stronger than the corresponding ones in Fig 1B, the ll1 peaks in Fig 1B would have completely disappeared if our filaments took the same C1 symmetry as those by Wu et al. The pattern of the layerlines in Fig 1B is similar to that of the C3 helical filament formed by the ASC Pyrin Domain (Fig. 1C), which was recently published by Lu et al (Cell 156, 1103). In particular, the ASC Pyrin filaments also showed clear layerline 1 peaks, similar to our MAVS CARD filaments. Like the MAVS CARD domain, Pyrin domain is a member of the Death Domain superfamily. Recent studies have demonstrated that both MAVS CARD and ASC Pyrin domains behave as prion-like filaments (Cai et al. 2014, Cell 156,1207; Lu et al. 2014, Cell 156, 1103). Our analyses suggest that MAVS CARD and ASC Pyrin domains both adopt C3 symmetry when they form prion-like helical filaments.\n\nTo independently evaluate our raw data, we invited two experts from two different institutions to calculate the power spectra from a good set of boxed segments using their own software packages. Their results (Fig. 1D and 1E) were essentially the same as ours as shown in Fig. 1B, and the peaks in Layerline 1 can be recognized despite that the signal-to-noise ratio in our raw data was lower than that of Wu et al. Layerline 1 reflects the long-range property of the filaments, and this difference in the power spectra is critical in building the filament model.\n\nTherefore, three different groups independently arrived at the same power spectrum based on the raw dataset of our MAVS CARD filament, which was presented in our eLife paper. This power spectrum is very similar to that of ASC Pyrin filaments, which exhibit the C3 symmetry, and is clearly different from the power spectrum of the renatured MAVS CARD filament presented in the Wu et al paper, which shows C1 symmetry. In particular, the two clear differences in power spectra (layerlines 1 and 9) indicate that the two filaments have different symmetry properties.\n\n3) Because of the potential ambiguity in the Fourier-Bessel indexing, especially with the limited resolution of our data, we had tested systematically the C1, C2, C3 and C4 symmetry against our data. Our test of C1 around the parameter sets of (102Âº, 5.0-5.6 Ã…) did not show any sign of convergence. Now with the symmetry property of Wu et al, we compared our test results with their C1 symmetry (101.1Âº, 5.13 Ã…; black â€œ+â€ in Fig. 2A). Our tests started from different positions (highlighted by black circles), but none of them converged to the symmetry parameters of Wu et alâ€™s filament. More strikingly, even when we started from the C1 parameters (brown squares), the refinement went away immediately, suggesting that our data are inconsistent with the C1 symmetry of Wu et al.\n\nIn contrast, when we tested the convergence around the C3 parameters (53.55Âº, 16.75 Ã…; black + in Fig 2B), all tests from different starting points (highlighted by black circles in Fig. 2B) converged to similar parameters. The convergence tolerated at least 10% perturbation in axial rise and at least ~3% perturbation in rotational angles. These tests argue strongly that our data are better described by the C3 symmetry, not the C1 symmetry as in the filaments by Wu et al.\n\nAs a control test, the C3 symmetry is a combination of a C1 and a three-fold rotation around the helical axis. It means that a C1 search around (53.55Âº, 16.7 Ã…) against our data should lead to a stable solution, and the obtained 3D reconstruction should have apparent three-fold rotational symmetry. These two predictions were indeed true. The C1 search of our data starting with a cylindrical volume stabilized at (53.02Âº, 16.60 Ã…), which are within 2% error range around the C3 parameters, and the obtained map showed an apparent three fold symmetry. This control test verified two important aspects of our analysis: 1) our C1 search procedure is reliable; 2) the three-fold symmetry appears intrinsic to our data at the current resolution.\n\n4) Some of the raw cryoEM images of our filaments taken at a high defocus level showed lower density in the middle (for example, red arrows in Fig. 3A in the attached PDF). This feature suggests that there might be a pore in the middle of the filament. As a test, we randomly assigned angles to the images aligned against a cylinder, and obtained an averaged reconstruction. The density showed a pore (~16 Ã… in diameter) in the middle (Fig. 3B). This feature was again observed in the final reconstruction of our filament, and was seen in the electron tomogram of the filaments.\n\nIn contrast, the C1 filament, whose two views are shown (Fig. 3C) as in the Wu et al Molecular Cell paper, is densely packed, and does not harbor a sizeable pore in the middle.\n\nThus, the presence of the central pore in our filament is compatible with the C3 reconstruction model presented in our eLife paper, but not the C1 model in the Wu et al paper.\n\n5) In the Mol Cell paper by Wu et al, the proximity of W56 and R43 was used as a piece of key evidence to discriminate the two structural models (Fig. 2B in the Wu et al paper). Even though the electrostatic compensation between W56R and R43D/E worked for in vitro filament formation by the seeding/ denaturation/ renaturation process (Fig. 2D in the Wu et al paper), the IFNbeta reporter assay showed that these double mutations did not rescue the functions of either human or duck MAVS in cells (Fig. S2D and S2F in the Wu et al paper). Thus, this experiment does not support that the C1 filament is a functional form capable of triggering MAVS signaling in vivo.\n\n6) It is well known that it is difficult to detect a stable complex between RIG-I and MAVS, and there is strong evidence to suggest that RIG-I dissociates from MAVS after MAVS starts to form the filaments; this allows RIG-I to promote MAVS polymerization in a â€˜catalyticâ€™ manner. In our model, the symmetry mismatch between the RIG-I tandem CARD tetramer and the C3 helix suggests that after the initiation of the filament, the RIG-I active complex may fall off, instead of staying bound. The molecular imprinting mechanism suggested by Wu et al was based in part on the complex of the RIG-I tandem CARD and the MAVS CARD mutant that were covalently linked together. The C1 model suggests that RIG-I CARDs might stay bound to MAVS after the filament formation.\n\nIn summary, our extensive analyses reconfirm that the filaments formed by the native wild-type MAVS CARD domain exhibit C3 symmetry, which is structurally different from the C1 filaments prepared through a denaturation-renaturation process as presented in the Wu et al paper. Future research should determine whether one or both of these structures represent the physiologically activated form of MAVS in vivo.\n\nFigure legends:\n\nFigure 1. Comparison of the summed power spectra from the raw data A) Power spectrum of the C1 filaments in Wu et al. (Mol Cell) B) Power spectrum of the C3 filaments in Xu et al. (eLife) C) Power spectrum of one of the sorted bin of images from the Cell paper by Lu et al 2014 on the ASC PYD filament, which shows C3 symmetry. Sorting sharpened the layerline 1 here. The pattern is similar to B, not A, because the layerline 1 peaks are almost invisible in A. D & E) Power spectra calculated by two independent scientists respectively; these spectra are consistent with B, supporting the model presented in the eLife paper by Xu et al.\n\nFigure 2. Convergence test of C1 versus C3 against the image data from filaments made of native wild-type MAVS CARD proteins A) C1 test. The starting points, highlighted by black circles, are all within the vicinity of the symmetry of the Molecular Cell paper by Wu et al (101.1Âº, 15.1Ã…). The refinements all led to further deviation from the expected C1 symmetry parameters. B) C3 test. The starting points are highlighted by black circles surrounding the first markers. The refinements all converged to the vicinity of (53.55Âº and 16.75 Ã…).\n\nFigure 3. A central pore in the filament made from native MAVS CARD proteins A) The cryoEM image from a short segment of the CARD filament showing the middle (red arrows) of the filament has lower density than the two edges. B) A reconstruction calculated from filaments that were aligned against a cylinder reference volume and assigned with random rotational angles along helical axes. Because of the averaging the volume is solid, except that it has lower density in the middle portion. The current threshold level gives a mass that is about 1.5 folds of the final reconstruction. At this level, there is a 16 Ã… hole in the middle, consistent with a lower density level in the middle of the raw images in A. C) Two views of the C1 filament from the Molecular Cell paper by Wu et al. The right panel is in the same orientation as in B. The densely packed units should display a higher density in the central part of the filament, which is distinct from the filament shown in A.\n\n\n<a href=\"https://uploads.disquscdn.com/images/1e5b6a8f7d686ad8583f41c5ea7c0a8731bb0b34f489370b6026ae6819bda891.jpg\">https://uploads.disquscdn.com/images/1e5b6a8f7d686ad8583f41c5ea7c0a8731bb0b34f489370b6026ae6819bda891.jpg</a>  <a href=\"https://uploads.disquscdn.com/images/0a865cd1e8c92a92047cc97f3ba8852bfe486d2f5d64b9dde0398c6b83e87109.jpg\">https://uploads.disquscdn.com/images/0a865cd1e8c92a92047cc97f3ba8852bfe486d2f5d64b9dde0398c6b83e87109.jpg</a>  <a href=\"https://uploads.disquscdn.com/images/3c954ee3404c1de29fbeaacc5c47d539cb5756397ad16f5b80416598d26c57c5.jpg\">https://uploads.disquscdn.com/images/3c954ee3404c1de29fbeaacc5c47d539cb5756397ad16f5b80416598d26c57c5.jpg</a>",
        ];
    }

    /**
     * @test
     * @dataProvider providerLineBreaks
     */
    public function it_will_preserve_line_breaks($raw_message, $expected)
    {
        $actual = convert_raw_message_to_markdown($raw_message);
        $this->assertContains($expected, $actual);
    }

    public function providerLineBreaks()
    {
        yield 'single line-break' => [
            "<strong>Comment on Version 2</strong>\nThe following sentence and citation to Bio-protocol (Müller and Münch, 2016) was added to the  Materials and methods:\n\nFinally, we tested whether the molecular tweezer inhibited semen-mediated infection enhancement, as described (Müller and Münch, 2016).\n\nThe following citation has been added to the Reference list:\nMüller, JA and Münch, J. (2016). Reporter assay for semen-mediated enhancement of HIV-1 infection. Bio-protocol 6(14): e1871. http://dx.doi.org/10.21769/BioProtoc.1871",
            "**Comment on Version 2**\n\nThe following sentence and citation",
        ];
        yield 'mulitple line-breaks' => [
            "<b>On the Physical Limits of Magnetogenetics</b>\n\nIn a recent paper, Markus Meister comments on data published by our groups (and a third employing a different approach) [1] showing that cells can be engineered to respond to an electromagnetic field [2-4]. Based on a set of theoretical calculations, Meister asserts that neither the heat transfer nor mechanical force created by an electromagnetic field interacting with a ferritin particle would be of sufficient magnitude to gate an ion channel and then goes on to question our groups\u2019 findings altogether.\n\nOne series of papers (from the Friedman and Dordick laboratories) employed four different experimental approaches in cultured cells, tissue slices and animals <i>in vivo</i> to show that an electromagnetic field can induce ion flow in cells expressing ferritin tethered to the TRPV1 ion channel [2,3]. This experimental approach was validated <i>in vitro</i> by measuring calcium entry, reporter expression and electrophysiological changes in response to a magnetic field. The method was validated <i>in vivo</i> by assaying magnetically induced changes in reporter expression, blood glucose and plasma hormones levels, and alterations in feeding behavior in mice. \n\nThese results are wholly consistent with those in an independent publication (from the Guler and Deppmann laboratories) in which the investigators fused ferritin in frame to the TRPV4 ion channel [4]. In this report, magnetic sensitivity was validated <i>in vitro</i> using calcium entry and electrophysiological responses as outputs. Additionally, <i>in vivo</i> validation was demonstrated by analyzing magnetically induced behaviors in zebrafish and mice, and through single unit electrophysiological recordings. \n\nIn his paper, Meister incorrectly states our collective view on the operative mechanism [1]. While we are considering several hypotheses, we agree that the precise mechanism is undetermined. Lastly, although mathematical calculations can often be used to model biologic phenomena when enough of the relevant attributes of the system are known, the intrinsic complexity of biologic processes can in other instances limit the applicability of purely theoretical calculations [5]. It is our view that mathematical theory needs to accommodate the available data, not the other way around. We are thus surprised that Meister would stridently question the validity of an extensive data set published by two independent groups (and a third using a different method) without performing any experiments. However, we too are interested in defining the operative mechanism(s) and welcome further discussion and experimentation to bring data and theory into alignment.\n\n<br>\n\n<b>Jeffrey Friedman*</b>, Sarah Stanley, Leah Kelly, Alex Nectow, Xiaofei Yu, Sarah F Schmidt, Kaamashri Latcha\n<i>Department of Molecular Genetics, Rockefeller University</i>\n\n<b>Jonathan S Dordick*</b>, Jeremy Sauer\n<i>Department of Chemical and Biological Engineering, Rensselaer Polytechnic Institute</i>\n\n<b>Ali D G\u00fcler*</b>, Aarti M Purohit, Ryan M Grippo\n<b>Christopher D Deppmann*</b>, Michael A Wheeler\n<b>Sarah Kucenas*</b>, Cody J Smith\n<i>Department of Biology, University of Virginia</i>\n\n<b>Manoj K Patel*</b>, Matteo Ottolini, Bryan S Barker, Ronald P Gaykema\n<i>Department of Anesthesiology, University of Virginia</i>\n\n<b>*Laboratory Heads in Bold Lettering</b>\n\n<br>\n\n<b>References</b>\n1. Meister, M, Physical limits to magnetogenetics. eLife, 2016. 5. http://dx.doi.org/10.7554/eLife.17210\n\n2. Stanley, SA, J Sauer, RS Kane, JS Dordick, and JM Friedman, Corrigendum: Remote regulation of glucose homeostasis in mice using genetically encoded nanoparticles. Nat Med, 2015. 21(5): p. 537. http://dx.doi.org/10.1038/nm0515-537b\n\n3. Stanley, SA, L Kelly, KN Latcha, SF Schmidt, X Yu, AR Nectow, J Sauer, JP Dyke, JS Dordick, and JM Friedman, Bidirectional electromagnetic control of the hypothalamus regulates feeding and metabolism. Nature, 2016. 531(7596): p. 647-50. http://dx.doi.org/10.1038/nature17183\n\n4. Wheeler, MA, CJ Smith, M Ottolini, BS Barker, AM Purohit, RM Grippo, RP Gaykema, AJ Spano, MP Beenhakker, S Kucenas, MK Patel, CD Deppmann, and AD Guler, Genetically targeted magnetic control of the nervous system. Nat Neurosci, 2016. 19(5): p. 756-61. http://dx.doi.org/10.1038/nn.4265\n\n5. Laughlin, RB and D Pines, The theory of everything. Proc Natl Acad Sci U S A, 2000. 97(1): p. 28-31. http://dx.doi.org/10.1073/pnas.97.1.28",
            "_Department of Anesthesiology, University of Virginia_\n\n**\*Laboratory Heads in Bold Lettering**\n\n**References**\n\n(1) Meister",
        ];
        yield 'line-break with preceding spaces' => [
            "<p>It is great to see that elife will be promoting data sharing upfront. This is an obvious sibbling of open access publications.</p><p>Many datasets cannot be encoded in standard formats and catered for in dedicated specialised repositories. Therefore, efforts like Dryad are very important. <br>But I am surprised to see no mention of existing public repositories for sharing data in the life sciences, e.g. Genbank/ENA, PDB, Geo/ArrayExpress, PRIDE, BioModels etc. What will be the policy of elife in that matter?</p><p>And I can only support Susanna when it comes to standards. Data sharing is one thing. But data re-use is another. And the later comes from the use of standards, official or de facto.</p>",
            "Therefore, efforts like Dryad are very important.\n\nBut I am surprised",
        ];
        yield 'preserve single line-breaks after references' => [
            "[Originally posted on 31 December 2014]\n\nGilbert et al. provide a helpful historical perspective to some aspects of our recent article on mechanisms of membrane pore formation by the cholesterol-dependent cytolysin (CDC) suilysin (<a href=\"http://dx.doi.org/10.7554/eLife.04247\">http://dx.doi.org/10.7554/eLife.04247</a>). Indeed, the interested reader is encouraged to study the references they provide, several of which are also cited in our paper.\n\nWe challenge, however, their implicit suggestion that we have insufficiently recognized past contributions by others. In our paper, the representation of the past literature is necessarily selective and not exhaustive – this being a research article and not a review. This selection takes into account both precedence of previous claims and the strength of the evidence by which they were supported. Besides the observation that this selection is by default subjective and therefore cannot be identical to that of Gilbert et al., a few specific points merit further consideration.\n\nFirstly, Gilbert et al. erroneously identify the kinetic model in our work with their previously proposed model (Gilbert, 2005, Sonnen et al., 2014). As described in Gilbert, 2005: “incomplete prepore structures are metastable and membrane insertion occurs when monomer recruitment slows due to depletion of local concentrations of toxin; that is, that the population of pore sizes formed is the result of a competition between prepore growth and pore insertion”. However, our Figure 5B,C shows that oligomeric populations of suilysin are identical for membrane incubations with the prepore-locked suilysin (unaffected by pore insertion) and with the unlocked, pore-forming suilysin. In other words, the suilysin assembly has been arrested and determined before pore insertion (and is not arrested by pore insertion). In our kinetic model, the assembly process is arrested by the depletion of a finite stock of suilysin monomers.\n\nGilbert et al. also appear to confuse the conclusions above with the more general – and previously made (Hotze et al., 2001) – observation that there is “a definite transition between the prepore state and the pore”. In fact, the perfringolysin data recorded by Hotze et al., 2001 show that the prepore-to-pore transition is fast compared to a preceding phase that includes CDC diffusion and binding to the membrane, and nucleation and formation of prepore assemblies on the membrane. This observation about insertion kinetics is quite distinct from (though not inconsistent with) our mechanistic conclusion that the prepore-to-pore transition requires a concerted conformational change of the whole assembly (based on experiments with mixtures of wild-type and disulphide-locked suilysin, presented in our Figure 4-figure supplement 3), which suggests a cooperative insertion of its subunits.\n\nRegarding the beautiful AFM experiments by Czajkowsky et al., 2004, they indeed revealed arcs with a height that corresponds to the membrane-inserted pore state of CDCs. However, those experiments were carried out with AFM probes of insufficient sharpness and aspect ratio to verify the membrane in the arc lumen, and thus did not allow for any conclusions about membrane lesions.\n\nWhere Gilbert et al. in that context refer to a “tell-tale reduction in height”, the authors of that AFM study themselves have been somewhat more cautious (Hotze and Tweten, 2012): “Whether the incomplete rings can insert a partial β-barrel remains unknown: the available data suggests that they do not insert into the membrane and may be artifacts of the EM process, dead end complexes or simply a matter of partially inactive protein preparations.”\n\nThis recent comment also illustrates that the proposition of pore-forming CDC arcs, though old, was until now more speculative than Gilbert et al.'s comment appears to suggest, in spite of additional electrophysiology data and electron tomography data (which are explicitly mentioned and cited in our paper, Marchioretto et al., 2013 and Sonnen et al., 2014).\nWhile we appreciate the positive comments of Gilbert et al. on our work, we strongly believe that its strength is significantly more than confirming and quantifying a pre-existing model.\n\n<strong>References</strong>\n\nCzajkowsky, D.M., Hotze, E.M., Shao, Z.F., and Tweten, R.K. (2004). Vertical collapse of a cytolysin prepore moves its transmembrane beta-hairpins to the membrane. EMBO J. 23, 3206-3215.\n\nGilbert, R.J. (2005). Inactivation and activity of cholesterol-dependent cytolysins: what structural studies tell us. Structure 13, 1097-1106.\n\nHotze, E.M., and Tweten, R.K. (2012). Membrane assembly of the cholesterol-dependent cytolysin pore complex. BBA-Biomembranes 1818, 1028-1038.\n\nHotze, E.M., Wilson-Kubalek, E.M., Rossjohn, J., Parker, M.W., Johnson, A.E., and\nTweten, R.K. (2001). Arresting pore formation of a cholesterol-dependent\ncytolysin by disulfide trapping synchronizes the insertion of the transmembrane\nbeta-sheet from a prepore intermediate. J. Biol. Chem 276, 8261-8268.\n\nMarchioretto, M., Podobnik, M., Dalla Serra, M., and Anderluh, G. (2013). What planar lipid membranes tell us about the pore-forming activity of cholesterol-dependent\ncytolysins. Biophys. Chem. 182, 64-70.\n\nSonnen, A.F.P., Plitzko, J.M., and Gilbert, R.J.C. (2014). Incomplete pneumolysin oligomers form membrane pores. Open Biol. 4, 140044.",
            "2014).\n\nWhile we appreciate the positive comments of Gilbert et al. on our work, we strongly believe that its strength is significantly more than confirming and quantifying a pre-existing model.\n\n**References**\n\nCzajkowsky, D.M., Hotze, E.M., Shao, Z.F., and Tweten, R.K. (2004). Vertical collapse of a cytolysin prepore moves its transmembrane beta-hairpins to the membrane. EMBO J. 23, 3206-3215.\n\nGilbert, R.J. (2005). Inactivation and activity of cholesterol-dependent cytolysins: what structural studies tell us. Structure 13, 1097-1106.\n\nHotze, E.M., and Tweten, R.K. (2012). Membrane assembly of the cholesterol-dependent cytolysin pore complex. BBA-Biomembranes 1818, 1028-1038.\n\nHotze, E.M., Wilson-Kubalek, E.M., Rossjohn, J., Parker, M.W., Johnson, A.E., and\nTweten, R.K. (2001). Arresting pore formation of a cholesterol-dependent\ncytolysin by disulfide trapping synchronizes the insertion of the transmembrane\nbeta-sheet from a prepore intermediate. J. Biol. Chem 276, 8261-8268.\n\nMarchioretto, M., Podobnik, M., Dalla Serra, M., and Anderluh, G. (2013). What planar lipid membranes tell us about the pore-forming activity of cholesterol-dependent\ncytolysins. Biophys. Chem. 182, 64-70.\n\nSonnen, A.F.P., Plitzko, J.M., and Gilbert, R.J.C. (2014). Incomplete pneumolysin oligomers form membrane pores. Open Biol. 4, 140044.",
        ];
    }

    /**
     * @test
     * @dataProvider providerConvertUrls
     */
    public function it_will_convert_urls_to_markdown_links($markdown, $expected) {
        $actual = convert_urls_to_markdown_links($markdown);
        $this->assertContains($expected, $actual);
    }

    public function providerConvertUrls()
    {
        yield 'in brackets' => [
            "Yes. We feel this kind of functionality sits better in the web view. We feel that to get this kind of functionality into a pdf it is better to use tools such as the awesome Utopia Documents (<http://getutopia.com/)>.",
            "(<http://getutopia.com/>)",
        ];
        yield 'in pointy brackets' => [
            "Collective action by early-career researchers in Germany also let to the improvment of their working conditions. This is highlighted in this excellent ScienceCareers article <http://bit.ly/1yLPED6> also citing this eLife article. It is well worth a read as it nicely lays out the strategies and set backs of ECRs at the Max Planck Institutes. At the end they managed to improve their conditions with legally binding contracts! Good old politics, public pressure via the media and collective action made it all happen.",
            "ScienceCareers article <http://bit.ly/1yLPED6> also citing",
        ];
        yield 'url at end of message' => [
            "I consider revolutionary and courageous the work of Randy Schekman, as an epochal event that posterity will appreciate without any doubt. \"I am doing this work because I believe that journals need to be radically improved and we have the means to achieve this\", Randy Schekman states.In following, what accounts for the reason he is right. There is a general admission that CVD is the leading cause of death in western countries. Despite 26 papers of meine, I will send on request, on CAD (and thus CVD!) Inherited Real Risk, among them some MEDLINE, the majority, but not all, \"luxury Jornals\" have not yet accepted for publication an article of mine and my co-Authors on such a outstanding topic, although the IAS, International Society of Atherosclerosis, has invited me to send a COMMENTARY, published in 2009 on [www.athero.org](http://www.athero.org)! With great pleasure, therefore, I intend offering all the support of the International Society of Quantum Biophysical Semeiotics, and my personal, to Randy Schekman, because I approve and share his suggested four places to start. Ad Majora.\n\n<http://www.athero.org/>",
            "Ad Majora.\n\n<http://www.athero.org/>",
        ];
        yield 'short youtube url' => [
            "A video news release that explains the findings of this study is available to watch here: <https://youtu.be/ql6z0-aLVcA>\n\n<http://www.youtube.com/watch?v=ql6z0-aLVcA>",
            "to watch here: https://youtu.be/ql6z0-aLVcA",
        ];
        yield 'youtube url' => [
            "Carbon offsets are not a solution:\n\nhttp://www.nytimes.com/2009/11/18/science/earth/18offset.html\n\nBut as for what happens to tourism, that is an excellent question. Tourism is about 10% of global GDP. Notice the idea from John Shepherd cued in video.\n\nhttps://youtu.be/U4plQKk4obk?t=39m51s\n\nDecades from now, carbon-neutral fuels may be available for a new generation of aircraft, but for now, less flying is the best approach.\n\nhttp://www.youtube.com/watch?v=U4plQKk4obk",
            "cued in video.\n\nhttps://youtu.be/U4plQKk4obk?t=39m51s\n\nDecades from now, carbon-neutral fuels may be available for a new generation of aircraft, but for now, less flying is the best approach.",
        ];
        yield 'another youtube url' => [
            "Here is a cool animated description of this work, produced by the Instituto Gulbenkian de Ciência Outreach team.\n\n<https://www.youtube.com/watch?v=HlbMfAc9WWE&t=1s>\n\n<http://www.youtube.com/watch?v=HlbMfAc9WWE>",
            "team.\n\nhttps://www.youtube.com/watch?v=HlbMfAc9WWE&t=1s",
        ];
        yield 'image links' => [
            "**Cigarette smoke particulates, carbon black, and emphysema; a commentary**\n\nIshrat Chaudhuri1, Peter Morfeld2,3, Scott Crocker1, Yufanyi Ngiewih4, Len Levy5, Robert J. McCunney6,7\n\n1 Cabot Corporation, Billerica, Massachusetts, USA\n\n2 Institute for Occupational Epidemiology and Risk Assessment of Evonik Industries, Essen, Germany\n\n3 Institute and Policlinic for Occupational Medicine, Environmental Medicine and Preventive Research, University of Cologne, Germany\n\n4 Orion Engineered Carbons GmbH, Frankfurt am Main, Germany\n\n5 Cranfield University, UK\n\n6 Massachusetts Institute of Technology, Cambridge, Massachusetts, USA\n\n7 Brigham and Women\\u2019s Hospital, Boston, Massachusetts, USA\n\nAddress for Correspondence\n\nDr. Ishrat Chaudhuri\n\nCabot Corporation\n\n157 Concord Rd, Billerica, MA 01821\n\nUSA\n\nIshrat.chaudhuri@cabotcorp.com\n\nT +1 978 670 6965\n\nSummary\n\nYou and colleagues (1) published an interesting article in which they propose that cigarette smoke produces carbon black, and that carbon black is responsible for emphysema in smokers. The material identified by the authors as carbon black in the lungs of both smokers and laboratory mice exposed to cigarette smoke, however, is unlikely to be the same material as industrially-produced carbon black. We also review a mouse instillation study in which mice were dosed with industrially produced carbon black, and the study results were extrapolated to implicate adverse health effects in non-smoking workers handling industrial carbon black. The conclusion of the authors that \\u201cour findings underscore the need for all individuals and societies to minimize the production of \\u2026.industrial nCB\\u201d is a somewhat far-reaching statement, and we believe not justified based on the results of the study.\n\nIntroduction\n\nCarbon black (CB), a major industrial chemical, is used primarily as a reinforcing agent for rubber products and as a pigment. CB is a manufactured product that has been in commerce for over a century. In general, CB dusts in the workplace environment occur as aggregates or agglomerates (average diameter about 80 to 500 nm for aggregates of furnace blacks, agglomerates about 1 to 100+ \\u00b5m) (2). CB is nearly 99% pure carbon.\n\nYou et al (1) performed a study where they reportedly identified CB as being formed in cigarette smoke. The authors refer to CB in their study as nano particulate carbon black (nCB). Where CB is present in the workplace (CB production facilities and some downstream users), CB is present as aggregates or agglomerates, and there is no distinction between nCB (as described by You et al. (1)) and CB. You et al. (1) claim that CB formed in cigarette smoke is the cause of emphysema in smokers. They propose that CB accumulates in human myeloid dendritic cells in the lungs of a human smoker and in lung antigen presenting cells of mice chronically exposed to cigarette smoke. They also report that intranasal administration of industrial CB in mice induced emphysema in mouse lungs. They then extrapolate these results to industrial worker exposure to CB and suggest that these results emphasize the need to minimize the production of and exposure to industrial CB. However, the data presented by the authors do not support these conclusions, and we wish to comment on the following aspects of the paper:\n\n(1) The identification of black particles in emphysematous lungs as CB is not conclusive.\n\n(2) Cigarette smoke is unlikely to result in the formation of CB.\n\n(3) Mouse instillation study results cannot be extrapolated to represent exposure to CB industrial production workers and downstream users.\n\n(1) The identification of black particles in emphysematous lungs as CB is not conclusive\n\nThe analytical data provided in You et al. (1) identifying CB in lungs of human smokers and mice exposed to cigarette smoke does not distinguish CB from soot, which is more likely to be formed in cigarette smoke. The American Society for Testing and Materials (ASTM) has developed a protocol for distinguishing CB from soot and other environmental particulates (3); it makes the point that it can be difficult to distinguish CB from soot. The ASTM definitions of CB (ASTM (4)) and soot (ASTM (3)) are noted below:\n\n\\u201cCarbon black \\u2013 an engineered material, primarily composed of elemental carbon, obtained from the partial combustion or thermal decomposition of hydrocarbons, existing as aggregates of aciniform morphology which are composed of spheroidal primary particles which exhibit uniformity of primary particle sizes within a given aggregate and turbostratic layering within the primary particles.\\u201d\n\n\\u201cSoot - a submicron black powder generally produced as an unwanted by-product of combustion or pyrolysis. It consists of various quantities of carbonaceous and inorganic solids in conjunction with adsorbed and occluded organic tars and resins.\\u201d\n\nRegarding soot, ASTM (3) also states \\u201cThe carbonaceous portion also is colloidal and often has the aciniform morphology.\\u201d \[Aciniform is defined as \\u201cshaped like a cluster of grapes.\\u201d\]\n\nFigure 1(C) in You et al. (1) is a transmission electron micrograph (TEM) of the residual black material from digested human emphysema lung tissue. Although the authors identify this material as CB, it is more likely to be soot based on the shape. ASTM (3) states that \\u201cin carbon black aggregates, the dimensions of the particle necks are smaller than the diameter of the primary particles. Whereas, in some soots, the necks have similar dimensions to the primary particles, thereby making it difficult to distinguish one primary particle from the next.\\u201d In Figure 1(C), it is difficult to distinguish between the primary particles. In general, more than one micrograph is necessary to distinguish manufactured CB from adventitious soot. For example, see the following micrograph of diesel soot provided in ASTM (3); it looks very similar to Figure 1(C) in You et al. (1).\n\nFigure 1(D) in You et al. (1) shows a Raman spectrum of the black material isolated from the lungs of one individual with emphysema. The authors state \\u201cthe bifid spectral peaks between 1000 and 2000 cm-1 are the typical Raman signature for carbon black\\u201d. The two broad ill-defined peaks seen in this spectrum, however, are also typical of other amorphous forms of carbon, such as soot. For example, Niessner (5) includes the following figure (Figure 19 in 5), which is a Raman spectrum of spark-discharge soot. The shape and position of the two peaks are almost identical to Figure 1(D) in You et al. (1).\n\nFigure 1(I) in You et al. (1) shows Raman spectra from lung cells isolated from lungs of mice exposed to smoke for 4 months. It is important to note that the Raman spectra from mouse lung cells (SMK CD11c and SMK Mac) look different from the CB reference spectrum (CB Ref), in that the peaks are broader and less-defined compared to CB Ref.\n\nIn fact, You et al. (1) acknowledge that Raman spectroscopy or hyperspectral imaging cannot distinguish CB from soot. The main reasons provided by You et al. (1) for stating that CB is present in the lungs of smokers are two-fold: their assumption that the high temperatures in cigarette smoke favor the generation of CB rather than soot, and that manufactured CB intranasally administered to mice appeared to replicate emphysema. Both of these issues are discussed further.\n\n(2) Cigarette smoke is unlikely to result in the formation of CB\n\nThe pyrolytic reaction in cigarette smoke is much different than the carefully controlled temperature, pressure, residence time and quenching functions of a reactor designed to manufacture CB. Greater than 95% of all CB produced in the world is manufactured using the furnace black process (e.g., the CB types used as test materials in the You et al. (1) study - Monarch 1100, Vulcan 9A32 and Sterling NS1 - are all produced using the furnace black process). The furnace black process consists of atomizing preheated oil in a combustion gas stream that is formed by burning fuel in preheated air (6). CB is produced by the furnace black process with a reactor temperature ranging from approximately 1400oC to 1800oC (2). Residence time ranges from an order of 1 second for the lowest temperature to a few milliseconds for the highest temperature. After an appropriate residence time, water is added to stop the reaction. The temperature after water injection is typically in the 900 to 1200oC range (7). It is not reasonably possible to make CB in the furnace process at a temperature of 1100oC or less.\n\nYou et al. (1) state that the temperature in a burning cigarette is 850 to 920oC during active inhalation, decreasing to 700oC during the smoldering phase. This temperature is much lower than temperatures required to form CB in the furnace black process. Another noteworthy difference between a burning cigarette and a CB furnace reactor is the way the reaction is quenched. In the CB manufacturing process, CB is quenched with water ending in a reducing environment. However, cigarette smoke is quenched by an excess of cool air ending in an oxidizing environment. The different reaction processes between cigarette smoke and CB manufacturing makes it unlikely that the black carbonaceous material formed in cigarette smoke is equivalent to CB formed using the currently predominant manufacturing method.\n\n(3) Mouse instillation study results cannot be extrapolated to represent exposure to CB industrial production workers and downstream users.\n\nYou et al. (1) conducted an intranasal instillation study in mice whereby 0.5 mg of CB (Monarch 1100 from Cabot) was instilled twice a week for 6 weeks, resulting in a total instillation dose of 6 mg. The dose used in the mouse study is significantly higher than doses used in similar toxicology studies, and the high dose and mode of administration cannot reasonably be extrapolated to predict adverse health effects in CB production workers.\n\nA critical problem with the mouse instillation experiment in You et al. (1) is that one very high dose (0.5 mg instilled twice a week for 6 weeks resulting in a total dose of 6 mg) was used, instead of a range of doses, as is typical in toxicology studies, to evaluate a dose-response effect. The dose used is significantly higher than doses used in similar studies in the same strain of mice. Porter et al. (8) studied pulmonary toxicity caused by titanium dioxide, which is an insoluble substance found to cause pulmonary effects similar to CB. In this study, C57BL mice were given single doses of 0.002, 0.0075, 0.015 and 0.03 mg titanium dioxide through pharyngeal aspiration. These doses are orders of magnitude lower than the dose used by You et al. (1), and also the use of a range of doses allowed the evaluation of a dose-response effect. While the use of a high dose indicates that it may be possible to evoke certain adverse effects, the relevance and interpretation of such findings is not clear if more relevant exposure levels are not taken into account.\n\nThe authors exposed the mice to CB by bi-weekly intranasal instillations. ECETOC noted that; \\u201cA major concern regarding the use of intratracheal instillation is that the introduction of a large bolus dose of the test substance into the lung in a short period of time will overwhelm the normal lung response and defense systems to the extent that it may produce responses that are pathophysiological artefacts that would not be seen if the same dose was delivered over a longer period of time via an inhalation exposure. This can thus produce serious problems for the interpretation of both hazard identification and risk assessment.\\u201d (9) These same considerations also pertain to the bolus effect via the intranasal instillations.\n\nDiscussion\n\nThe analysis of the black material isolated from the lungs of one individual with emphysema does not allow a definitive conclusion as to its composition. The black material, which the authors attributed to nano carbon black, could likely be soot, an unwanted by-product of combustion. Cigarette smoke is most unlikely to form CB as the pyrolytic reaction that occurs in a cigarette is different from the carefully controlled temperature, pressure, residence time and quenching functions of a reactor designed to manufacture CB. The very high doses of CB used in the mouse intranasal instillation study do not allow a reliable extrapolation of risks to CB industrial workers and downstream users, who may be exposed to considerably lower levels of CB.\n\nCB is a manufactured product that has been in commerce for over a century. In our opinion, we respectfully propose that the conclusion of You et al. (1) that \\u201cour findings underscore the need for all individuals and societies to minimize the production of \\u2026.industrial nCB\\u201d is not justified based on the findings of their study, and the information and considerations we have provided.\n\n**Conflict of interest**\n\nIC and SC are employees of Cabot Corporation, and YN is an employee of Orion Engineered Carbons, both of which are carbon black manufacturing companies. IC, PM, YN, LL, and RJM are members of the Scientific Advisory Group to the International Carbon Black Association (ICBA, http://www.carbon-black.org). RJM serves as a medical consultant to Cabot Corporation and in this capacity is the Chair of the Scientific Advisory Group to ICBA. The ICBA is a scientific, non-profit corporation originally founded in 1977. The purpose of the ICBA is to sponsor, conduct and participate in investigations, research, and analyses relating to the health, safety, and environmental aspects of the production and use of carbon black.\n\n**References**\n\n(1) You R, Lu W, Shan M, Berlin JM, Samuel EL, et al. Nanoparticulate carbon black in cigarette smoke induces DNA cleavage and Th17-mediated emphysema. _eLife_ 2015;10.7554/eLife.09623 (2015).\n\n(2) McCunney RJ, Muranko HJ, Long CM, Hamade AK, Valberg, PA, Morfeld P. Patty\\u2019s Chapter 89, Carbon Black. Patty\\u2019s Toxicology, Sixth Edition. Volume 5, Edited by Eula Bingham and Barbara Cohrssen. Published 2012 by John Wiley & Sons, Inc (2012).\n\n(3) ASTM. Standard Practice for Sampling and Testing of Possible Carbon Black Fugitive Emissions or Other Environmental Particulate, or Both. Designation: D6602 \\u2013 03b (Reapproved 2010). ASTM International, 100 Barr Harbour Dr., P.O. box C-700 West Conshohocken, Pennsylvania USA (2010).\n\n(4) ASTM. Standard Terminology Relating to Carbon Black.Designation: D3053-15. ASTM International, 100 Barr Harbour Dr., P.O. box C-700 West Conshohocken, Pennsylvania USA\n\n(5) Niessner R. The Many Faces of Soot: Characterization of Soot Nanoparticles Produced by Engines. Angew. Chem. Int. Ed. 2014, 53, 12366 \\u2013 12379 (2014).\n\n(6) Wang MJ, Gray CA, Reznek SA et al. Carbon black. In: Kirk-Othmer Encyclopedia of Chemical Technology, New York, John Wiley & Sons, Vol 4, p 761\\u2013803 (2003).\n\n(7) Donnet J-B, Bansai RC, Wang M-J. Carbon Black: Science and Technology, Second Edition. CRC Press (1993).\n\n(8) Porter DW, Wu N, Hubbs AF, Mercer RR, Funk K, Meng F, Li J, Wolfarth MG, Battelli L, Friend S, Andrew M, Hamilton R Jr, Sriram K, Yang F, Castranova V, Holian A. Differential mouse pulmonary dose and time course responses to titanium dioxide nanospheres and nanobelts. Toxicol Sci. 2013 Jan;131(1):179-93. (2013).\n\n(9) ECETOC 2014. Poorly Soluble Particles/Lung Overload, Technical Report No. 122 ISSN-0773-8072-122 (Print); ISSN-2073-1526-122 (Online).\n\n**Figures**\n\nFigure 1. Transmission electron micrograph of diesel soot\[1\]\n\n\[1\] Reproduced, with permission, from ASTM D6602 - 03b(2010)e1 (Historical Version) Standard Practice for Sampling and Testing of Possible Carbon Black Fugitive Emissions or Other Environmental Particulate, or Both, \\u00a9 copyright ASTM International, 100 Barr Harbor Drive, West Conshohocken, PA 19428. A copy of the complete standard may be obtained from ASTM, www.astm.org.\n\nFigure 2. Raman spectroscopy of spark-discharge soot\[2\]\n\n\[2\] Reproduced with permission from Wiley-VCH. Page 12376 of Reference 5. \\u00a9 Copyright 2014. Wiley-VCH Verlag GmbH & Co. KGaA\n\n<https://cdn.elifesciences.org/annotations-media/2690502361-001-original.jpg>\n\n<https://cdn.elifesciences.org/annotations-media/2690502361-002-original.jpg>",
            "KGaA\n\n[![](https://cdn.elifesciences.org/annotations-media/2690502361-001-original.jpg)](https://cdn.elifesciences.org/annotations-media/2690502361-001-original.jpg)\n\n[![](https://cdn.elifesciences.org/annotations-media/2690502361-002-original.jpg)](https://cdn.elifesciences.org/annotations-media/2690502361-002-original.jpg)",
        ];
        yield 'abbreviated link to media file' => [
            "Figure 4 in the accepted manuscript PDF is displaying incorrectly due to a processing error; the correct version of Figure 4 can be viewed and downloaded here: [http://elifesciences.org/conte...](http://elifesciences.org/content/elife/suppl/2014/07/22/eLife.03357v1.DC1/eLife03357_Figure_4.jpg)",
            "Figure 4 in the accepted manuscript PDF is displaying incorrectly due to a processing error; the correct version of Figure 4 can be viewed and downloaded here: [![](http://elifesciences.org/content/elife/suppl/2014/07/22/eLife.03357v1.DC1/eLife03357_Figure_4.jpg)](http://elifesciences.org/content/elife/suppl/2014/07/22/eLife.03357v1.DC1/eLife03357_Figure_4.jpg)",
        ];
    }

    /**
     * @test
     * @dataProvider providerOtherExamples
     */
    public function it_can_handle_other_examples($raw_message, $formula, $expecteds)
    {
        $actual = convert_raw_message_to_markdown($raw_message, $formula);
        foreach ((array) $expecteds as $expected) {
            $this->assertContains($expected, $actual);
        }
    }

    public function providerOtherExamples()
    {
        yield '< just before integer' => [
            "That data qualities are excellent.  \n\nWe have used splicingcode table of <230,000 introns, which are 10-20% of the estimated human introns,  to analyze their datasets. We have identified about 4,500 fusion transcripts, from which the numbers of highly-recurrent fusion  have been identified.  One of the most intriguing loci is at the chromosome  15q24.  It has read-through fusion transcripts of SH3GL3|ADAMTSL3 and actively-inversion of ADAMTSL3|SH3GL3 if it is a somatic mutation,         or alternative-spliced if it is germline-inherited. We have identified 17 ADAMTSL3|SH3GL3 isoforms or inversion types.  They are also detected in some of their controls.  The following fusion transcripts in the table are some of the highly-recurrent fusion transcripts. If you are interested in some of these data, you can go to http://splicingcodes.com to request data described in detail. ",
            [],
            "We have used splicingcode table of <230,000 introns",
        ];
        yield 'preserve formula' => [
            "[Originally posted on 14 January 2016]\n\nWe are writing to comment on the manuscript by Craven entitled “Evaluation of Predictions of the Stochastic Model for Organelle Production Based on Exact Distributions”. Before responding to the points raised in the Craven manuscript, we would first like to thank Jeremy Craven for improving our paper (Mukherji and O’Shea, 2014) through his careful attention to our work.\n\nOverall, there are three points we wish to emphasize at the outset of our response:\n\n1) The exact solution to the organelle birth-death model presented in the Craven paper represents a genuine advance that we anticipate will be of great use in applying this theoretical framework to the quantitative analysis of organelle biogenesis in the future.\n\n2) We believe, however, there is very little disagreement between the solutions to the organelle birth-death model presented in the two manuscripts. In the cases of both the Golgi and the peroxisomes, the expressions for the Fano factors in the two papers are identical. In the case of the vacuole, as we show below, there is at worst a 20% disagreement in the measured and exactly calculated Fano factors and virtually no difference between the inferred and exact predictions of the vacuole copy number distribution. Particularly for the case of the peroxisome, any differences of opinion about the biological interpretation of the Fano factor thus cannot arise from any new knowledge gained from the exact solution.\n\n3) Given the above point, we believe that the disagreement over inferences about the mode of peroxisome biogenesis rests on semantic differences and interpretation of previously measured data from mutant yeast strains rather than the mathematical reanalysis presented in this paper.\n\nDetailed response to issues raised by Craven regarding the Mukherji and O’Shea manuscript:\n\n<strong>Section: Does the SMOP analysis imply equilibrium distributions?</strong>\n\nCraven’s point is well taken – we acknowledge that it is difficult to prove that cells have reached a steady state distribution of organelle abundances without dynamic data. However, given our demonstration that the distributions for cells arrested in S-phase and cells growing exponentially are the same, there is no reason to believe that our assumption that organelle numbers are in steady state is invalid. In addition, we are not aware of any evidence that suggests a relationship between replicative age (i.e., the number of times a cell has budded) and organelle abundance, except for extremely high replicative ages (in the neighborhood of 10+ divisions; however, these cells will make up such a small fraction of the population that they will not contribute measurably to our distributions).\n\n<strong>Section: Application of recurrence method to Golgi and vacuole models</strong>\n\nIn the interest of clarity, it is worth stating that our theoretical work does not predict that the vacuole data will follow a shifted Poisson distribution. We merely inferred that the shifted Poisson distribution was the correct distribution describing the vacuole distribution on the basis that: (1) the shifted Poisson yields a formula for the Fano factor, σ2/<n> = 1-1/<n>, that is identical to what we obtained from our approximate Fano factor formula in the limit of k_denovo = gamma = 0; and (2) the distribution appeared to describe the experimentally measured vacuole distribution well.\n\nConsidering that the truncated Poisson distribution can be derived from the underlying model of organelle biogenesis, we wish to highlight that there is at worst a 20% disagreement between the Fano factor we measure experimentally (Craven Figure 2, open circle data point) and the Fano factor expected theoretically from the truncated Poisson distribution. Indeed, if we overlay the prediction from the truncated Poisson distribution whose mean matches the experimentally measured mean vacuole copy number with the shifted Poisson distribution and the experimentally measured vacuole distribution, we find very good agreement among all three curves (see our graph below). The parameter value used to generate the truncated Poisson curve in Craven Figure 4 does not give the same mean as experimentally measured for vacuoles and thus is not relevant to the interpretation of the experimental data.\n\nCrucially, we predict and observe a non-intuitive quantitative feature of the fission/fusion balance: the Fano factor increases with increasing mean rather than decreases. This prediction is insensitive to whether the true underlying distribution is a truncated or shifted Poisson. Finally, we can think of no reason why the marginally stronger agreement of our data with the shifted Poisson distribution carries any special biological significance. If the shifted Poisson distribution were the result of a radically different model of vacuole biogenesis, then such a result could possibly cast doubt on our method, but that is not being argued here.\n\n<strong>Section: Evaluation of Fano factor for the boundary marking equal de novo and fission rates</strong>\n\nIt is worth emphasizing that, although we used an approximation to derive the Fano factor for peroxisome biogenesis, our equation for the Fano factor turned out to be the exact solution as confirmed in the Craven manuscript. Thus, it is impossible for there to be any disagreement between the two analyses in using the Fano factor to assign the relative contributions of fission and de novo synthesis to peroxisome copy numbers. Both the new analysis and our original analysis leads to the conclusion that a Fano factor of 2.4 allows one to infer that the peroxisome fission rate is 40% larger than the peroxisome de novo synthesis rate. It is the larger fission rate compared to the smaller de novo synthesis rate that we refer to when we use the term “dominant”; the difference of opinion here appears to be semantic. Furthermore, it is also worth emphasizing that what our analysis particularly highlighted was that peroxisome fission is the primary process explaining peroxisome proliferation in oleic acid-cultured cells. The statistical signature allowing us to make that inference was the increased Fano factor in the peroxisome abundance distribution measured from oleic acid-cultured cells compared to glucose-cultured cells; specifically, the calculated ratio of the fission rate to the de novo synthesis rate in oleic acid-cultured cells is 14 times the ratio of the fission rate to de novo synthesis rate in glucose-cultured cells.\n\n<strong>Section: Restrictions on the relationships between parameters in order for limiting distributions to exist</strong>\n\nIn the Craven paper it is stated that there are 2 ways for the Fano factor to get large when fusion is negligible: 1) either fission is much larger than de novo synthesis or 2) the fission rate constant and the first order decay rate constant are similar magnitude. However, these 2 cases appear to be 2 sides of the same coin. At steady state, the summed rates of production must equal the summed rates of decay. In the case of the organelle birth-death model where the rate of fusion is 0, this means that:\n\nk_denovo + k_fission<n> = γ<n>\n\nIf the rate constant k_fission is set to be similar to the rate constant γ, the fission rate and the first order decay rate also become similar (since both rates are just the rate constants times the mean). But this implies, due to the accounting identity that the rates of production and rates of decay must balance, that the rate of fission has become large compared to the rate of de novo synthesis. Intuitively when the fission and first order decay rate constants, and therefore rates, are similar then there isn't much of a role for the de novo synthesis rate to play in creating the steady state condition. Therefore, setting the fission rate constant and first order decay rate constant to be similar to each other necessarily implies that the fission rate is larger than the de novo synthesis rate at steady state and so there does not appear to be a mathematical difference between the 2 ways stated above to make the Fano factor large.\n\n<strong>Section: Discussion</strong>\n\nThe published data on the vps1 mutant grown in glucose (shown in Table 1 of Kuravi et al (2006)) indicate that there is a 30% drop in peroxisome copy number relative to the wild-type strain: from a wild-type mean of 1.62 peroxisomes per cell to a vps1 mutant strain mean of 1.15 peroxisomes per cell, which is only modestly larger than the 10% drop predicted by our analysis. Our own unpublished observations, using strains expressing yeast enhanced monomeric Citrine tagged with peroxisome targeting signal 1 to mark peroxisomes, are consistent with a 30% drop in mean peroxisome copy number per cell from 3 in wild-type cells to 2 in vps1 mutant cells. Given our relatively poor knowledge of the mechanisms of de novo peroxisome biogenesis, especially with regard to the factors promoting pre-peroxisomal vesicle production from the endoplasmic reticulum and with regard to whether peroxisome fission factors may also play a role in de novo biogenesis, this is an area that we believe demands more study.\n\nAs a final note, it is worth taking a step back and noting how substantial a simplification the lumped parameter model of organelle biogenesis that we used is compared to what is actually happening microscopically – it is almost certainly an oversimplification of what is going on. Even with those severe limitations, and even when re-examined in light of the exact solutions presented in the new manuscript by Craven, our observations agree quite well with the quantitative data. As exemplified by our conclusion that peroxisome fission underlies the peroxisome proliferation in oleic acid-cultured cells, the close match between theory and experiment lends credence to the concept that organelle abundance distributions can shed light on mechanistic principles underlying organelle biogenesis. We hope that both papers stimulate further work in this area.\n\nGraph to accompany comment\n\nhttps://uploads.disquscdn.com/images/32884c2464cddf1c4b68e3b7f5da3df6f9955ad910f2682f75b41b928008689f.jpg",
            [
                'k_denovo + k_fission<n> = γ<n>',
                'σ2/<n> = 1-1/<n>',
            ],
            [
                '`k_denovo + k_fission<n> = γ<n>`',
                '`σ2/<n> = 1-1/<n>`',
            ],
        ];
        yield 'escape pound sign' => [
            "[Originally posted 10 July 2015]\n\n<strong>Comment: The viruses of Tulloch et al. do not maintain constant codon pair frequencies, and do not distinguish dinucleotide bias from codon pair bias</strong>\n\nBruce Futcher (1#), Oleksandr Gorbatsevych (1), Sam H Shen (3), Charles B Stauft (1,2), Yutong Song (1), Bingyin Wang (1), Janet Leatherwood (1), Justin Gardin (1), Alisa Yurovsky (1), Steffen Mueller (2), Eckard Wimmer (1,2#)\n\n-----\n1) Dept of Microbiology and Molecular Genetics, Stony Brook University, NY 11790\n2) Codagenix, 25-108 Health Sciences Dr, Stony Brook, NY 11790\n3) Present address: Integrated DNA Technologies, Coralville, Iowa 52241\n# Corresponding authors: bfutcher@gmail.com / eckard.wimmer@stonybrook.edu\n-----\n\nTulloch et al. (Tulloch et al., 2014) write that \u201c...codon pair deoptimization is an artefact of increases in CpG/UpA dinucleotide frequencies\u201d. We believe there is an error in their approach, which invalidates this conclusion.\n\nCodon pair bias (Gutman and Hatfield, 1989) and CpG/UpA dinucleotide bias (Beutler et al., 1989, Rothberg and Wimmer, 1981) are two encoding biases. In mammals, codon pairs that have CpG or UpA at the codon-codon junction (i.e., xxC Gxx or xxU Axx codon pairs, C3G1, U3A1) are \u201crare\u201d (i.e., less frequent than expected), as are the dinucleotides CpG and UpA. However, these two biases are not independent \u2013 it is not obvious whether these codon pairs are rare because the dinucleotides are rare, or whether the dinucleotides are rare because the codon pairs are rare, or whether both phenomena are a reflection of some other unknown force.\n\nWe and others have shown that when mammalian viruses are recoded to increase the frequency of very rare codon pairs, this attenuates the virus (Coleman et al., 2008). Because many of the rarest codon pairs in mammals have junctional CpG or UpA, and because of the mathematical linkage between the phenomena, these recoded viruses inevitably have increased frequencies of CpG and UpA dinucleotides. But it is difficult to say what is cause and what is effect. No mechanism for attenuation is known, and the real mechanism of attenuation may not be well-described by either of the terms \u201ccodon pair bias\u201d or \u201cdinucleotide bias\u201d. Our view is that neither term should be taken too seriously, and that a molecular understanding of mechanism is more important than the nomenclature.\n\nTulloch et al. (Tulloch et al., 2014) tried to distinguish experimentally between the \u201ccodon pair\u201d and the \u201cdinucleotide\u201d points of view, to see which was causing attenuation. But in evaluating their conclusion, it is essential to understand exactly what it is that they have done. The idea behind their experiment was to recode the \u201cMin-E\u201d version of echovirus 7 (E7) virus so as to keep codon pair frequencies constant, but increase CpG and UpA frequencies, and see whether the resulting test viruses (Min-U and Min-H) were attenuated (which they were). However, it is surprisingly difficult to implement this apparently straight-forward plan. At most positions, CpG and UpA dinucleotides cannot be inserted, because change would lead to a change in the amino acid sequence, or to an unacceptably large change in codon usage. Many positions that can accept recoding with CpG and UpA dinucleotides are at the codon-codon junctions, where a change necessarily also leads to a change in codon pair frequency, so that the two frequencies would change co-ordinately, not independently. How did Tulloch et al. deal with this problem?\n\nOur inspection of the sequences of the viruses Min-E, U, and H shows that about half of the new CpG/UpA dinucleotides were at codon-codon junctions, and these introduced many very rare codon pairs. That is, in the two test viruses Min-U and Min-H, the very rare C3G1 and U3A1 codon pairs are co-ordinately increased with CpG/UpA dinucleotide frequencies (see Figure 1 below); in fact, the rare codon pairs increase in frequency somewhat faster than the dinucleotides. Because of the co-ordinate increases, the attenuation of the two test viruses is consistent with both the \u201cdinucleotide\u201d and the \u201ccodon pair\u201d point of view, and so the hypotheses are not at all distinguished by these viruses.\n\nBut in contrast with our characterization of Min-E, U, and H in Figure 1, Tulloch et al. (Tulloch et al., 2014) made multiple claims that codon pair frequencies were kept constant in their Min-E, U, and H constructs (e.g., Min-H \u201chas the same CP frequencies as Min-E\u201d). We believe these statements are incorrect and misleading. Inspection reveals that Tulloch et al. have recoded these two viruses to also add, at other positions, other codon pairs that are over-represented (i.e., more frequent than expected) (see Figure 1 of (Futcher et al., 2015)). One can describe under-represented codon pairs as having a negative codon pair score and over-represented codon pairs as having a positive codon pair score (Coleman et al., 2008), and, because of the added positive codon pairs, if one takes all the codon pair scores in Min-U and Min-H and averages them, then that average is similar to the average in Min-E. That is, compared to Min-E, viruses Min-U and Min-H have an increased frequency of both negative (under-represented) and positive (over-represented) codon pairs, so as to achieve a constant average score. But this is not what Tulloch et al. say \u2014 they do not say that the average score is maintained \u2014 instead they say that frequencies are maintained, and this is not correct, and is not at all the same thing. There is no evidence that either the average score, or the number of over-represented codon pairs, has any functional significance (although we and others do use the average score as a book-keeping device when changing the frequency of under-represented codon pairs (Coleman et al., 2008)). Instead, attenuation is likely due to the number of very under-represented codon pairs. This number is significantly raised in both of the test viruses, along with the CpG and UpA dinucleotides. Thus the viruses constructed fail to test the hypothesis, and the conclusion of Tulloch et al. is unwarranted.\n\nIn their closing remarks Tulloch et al entertain a potential danger should codon pair deoptimized RNA viruses that relative to wild type viruses contain hundreds of silent nucleotide changes be used as vaccines. They ignore that such vaccine candidates, originally investigated only in tissue culture cells (Coleman et al., 2008; Tulloch et al., 2014) will undergo multiple years-long testing in animals and humans for safety, efficacy and genetic stability. Significantly, studies with codon pair deoptimized RNA viruses in tissue culture cells may not yield matching results in experimental animals (Shen et al., 2015).\n\n-----\n\n<strong>References</strong>\n\nBEUTLER, E, GELBART, T, HAN, JH, KOZIOL, JA & BEUTLER, B. 1989. Evolution of the genome and the genetic code: selection at the dinucleotide level by methylation and polyribonucleotide cleavage. Proc Natl Acad Sci U S A, 86, 192-6.\n\nCOLEMAN, JR, PAPAMICHAIL, D, SKIENA, S, FUTCHER, B, WIMMER, E & MUELLER, S. 2008. Virus attenuation by genome-scale changes in codon pair bias. Science, 320, 1784-7.\n\nFUTCHER, B, GORBATSEVYCH, O, SHEN, SH, STAUFT, CB, SONG, Y, WANG, B, LEATHERWOOD, J, GARDIN, J, YUROVSKY, A, MUELLER, S & WIMMER, E. 2015. Reply to Simmonds et al.: Codon pair and dinucleotide bias have not been functionally distinguished. Proc Natl Acad Sci U S A.\n\nGUTMAN, GA & HATFIELD, GW. 1989. Nonrandom utilization of codon pairs in Escherichia coli. Proc Natl Acad Sci U S A, 86, 3699-703.\n\nROTHBERG, PG & WIMMER, E. 1981. Mononucleotide and dinucleotide frequencies, and codon usage in poliovirion RNA. Nucleic Acids Res, 9, 6221-9.\n\nSHEN, SH, STAUFT, CB, GORBATSEVYCH, O, SONG, Y, WARD, CB, YUROVSKY, A, MUELLER, S, FUTCHER, B & WIMMER, E. 2015. Large-scale recoding of an arbovirus genome to rebalance its insect versus mammalian preference. Proc Natl Acad Sci U S A, 112, 4749-54.\n\nTULLOCH, F, ATKINSON, NJ, EVANS, DJ, RYAN, MD & SIMMONDS, P. 2014. RNA virus attenuation by codon pair deoptimisation is an artefact of increases in CpG/UpA dinucleotide frequencies. eLife, 3, e04531.\n\n-----\n\nFigure 1. Rare codon pairs and CpG/UpA dinucleotides are increased co-ordinately in the viruses of Tulloch et al.\n\nhttps://uploads.disquscdn.com/images/c4da4874752e6ee62725d97cab61f6f58d3e8b2975902d859336f730d1760b61.jpg",
            [],
            "Iowa 52241\n\n\# Corresponding authors: bfutcher@gmail.com / eckard.wimmer@stonybrook.edu\n\n-----",
        ];
        yield 'escape dollar sign' => [
            "<p>This is such a timely article. Indeed, the current system is pathological. Tail wagging the dog. Instead of rewarding scientific advancement, the system self-reinforces chasing bigger grants to build bigger labs so that even bigger grants can be obtained. Universities get their fat cut from every grant and promote the $$$ chase. The system turns scientist into part-time grant writers, part-time businessmen. Not only it wastes enormous amounts of time of the very scientists it is supposed to support but also goes to the very heart of why one becomes a scientist - to discover new knowledge, to satisfy one's curiosity - and poisons it with the chase for big fat $$$.</p>",
            [],
            'the $ chase. The system turns scientist into part-time grant writers, part-time businessmen. Not only it wastes enormous amounts of time of the very scientists it is supposed to support but also goes to the very heart of why one becomes a scientist - to discover new knowledge, to satisfy one\'s curiosity - and poisons it with the chase for big fat $.',
        ];
        yield 'hyphen at start of line' => [
            "<p>Thanks Mark for this question -</p><p>In this survey, we asked - What factors are important to you when selecting a journal to publish in? For this, we gave all those option mentioned in figure 2.</p><p> The last option was \u2013 \u2018other\u2019 - If there is any other factor, which is not mentioned among the choices then please specify this 'Other' factor. We also asked them to comment on the relevance of this other factor. Some of the other factors in the comments were -</p><p>- Prefer journals where editors are practicing scientists, rather than \"professional\"editors.<br>- If the paper is accepted only based on it being technically sound, and not for the perceived future impact, e.g. PLoS ONE, F1000Research. The authors and reviewers should be unknown.<br>- Listed in Web of Science.<br>- Ability to specify competing reviewers whom should not review the manuscript, due to fear of being shot down or scooped due to competition.<br>- Ability to make (preliminary version of the) paper available on personal institute homepage.<br>- Review Process is important e.g.  EMBO\u2019s transparent publishing process. In my field (computer science - verification/program analysis/type systems), journal publications are less important than good conference publications.<br>- I submit my article where boss says!.<br>- What my supervisors think, where I should publish.<br>- Speed of review procedure (not to be confused with speed of publication).<br>- Relevance to work being published.</p>",
            [],
            "-\n\n\- Prefer journals where editors are practicing scientists, rather than \"professional\"editors.\n\n\- If the paper",
        ];
    }
}
